src/components/Block/SpaceFormProfile.vue

```diff
@@ -16,67 +16,79 @@ const props = withDefaults(
 const emit = defineEmits<{
   (e: 'errors', value: any);
   (e: 'pick', field: any);
+  (e: 'no-network');
 }>();
 
 const availableNetworks = enabledNetworks.map(id => ({
   id,
   name: getNetwork(id).name
 }));
 
-const definition = {
-  type: 'object',
-  title: 'Space',
-  additionalProperties: false,
-  required: ['name', 'walletNetwork', 'walletAddress'],
-  properties: {
-    name: {
-      type: 'string',
-      title: 'Name',
-      minLength: 1,
-      examples: ['Space name']
-    },
-    description: {
-      type: 'string',
-      format: 'long',
-      title: 'About',
-      examples: ['Space description']
-    },
-    externalUrl: {
-      type: 'string',
-      title: 'Website',
-      examples: ['Website URL']
-    },
-    github: {
-      type: 'string',
-      title: 'GitHub',
-      examples: ['GitHub handle']
-    },
-    twitter: {
-      type: 'string',
-      title: 'Twitter',
-      examples: ['Twitter handle']
-    },
-    discord: {
-      type: 'string',
-      title: 'Discord',
-      examples: ['Discord handle']
-    },
-    walletNetwork: {
-      type: 'string',
-      enum: enabledNetworks,
-      options: availableNetworks,
-      title: 'Treasury network'
-    },
-    walletAddress: {
-      type: 'string',
-      format: 'address',
-      title: 'Treasury address',
-      examples: ['0x0000…']
+const definition = computed(() => {
+  return {
+    type: 'object',
+    title: 'Space',
+    additionalProperties: false,
+    required: ['name', 'walletNetwork', 'walletAddress'],
+    properties: {
+      name: {
+        type: 'string',
+        title: 'Name',
+        minLength: 1,
+        examples: ['Space name']
+      },
+      description: {
+        type: 'string',
+        format: 'long',
+        title: 'About',
+        examples: ['Space description']
+      },
+      externalUrl: {
+        type: 'string',
+        title: 'Website',
+        examples: ['Website URL']
+      },
+      github: {
+        type: 'string',
+        title: 'GitHub',
+        examples: ['GitHub handle']
+      },
+      twitter: {
+        type: 'string',
+        title: 'Twitter',
+        examples: ['Twitter handle']
+      },
+      discord: {
+        type: 'string',
+        title: 'Discord',
+        examples: ['Discord handle']
+      },
+      walletNetwork: {
+        type: ['string', 'null'],
+        enum: [...enabledNetworks, null],
+        options: [...availableNetworks, { id: null, name: 'No treasury' }],
+        title: 'Treasury network',
+        nullable: true
+      },
+      walletAddress: {
+        type: props.form.walletNetwork === null ? 'null' : 'string',
+        format: 'address',
+        title: 'Treasury address',
+        examples: ['0x0000…'],
+        hidden: props.form.walletNetwork === null
```

Comment:
I wonder if it would make sense to just remove `hidden` and check for `type === null`.

Reply:
I think it is more explicitly and we can use it somewhere else when we need to hide form fields based on some condition.